pushed_authorize_request_handler.go

```diff
@@ -0,0 +1,63 @@
+package fosite
+
+import (
+	"context"
+	"errors"
+	"net/http"
+
+	"github.com/ory/fosite/i18n"
+	"github.com/ory/x/errorsx"
+)
+
+// NewPushedAuthorizeRequest validates the request and produces an AuthorizeRequester object that can be stored
+func (f *Fosite) NewPushedAuthorizeRequest(ctx context.Context, r *http.Request) (AuthorizeRequester, error) {
+	request := NewAuthorizeRequest()
+	request.Request.Lang = i18n.GetLangFromRequest(f.Config.GetMessageCatalog(ctx), r)
+
+	if r.Method != "" && r.Method != "POST" {
+		return request, errorsx.WithStack(ErrInvalidRequest.WithHintf("HTTP method is '%s', expected 'POST'.", r.Method))
+	}
+
+	if err := r.ParseMultipartForm(1 << 20); err != nil && err != http.ErrNotMultipart {
+		return request, errorsx.WithStack(ErrInvalidRequest.WithHint("Unable to parse HTTP body, make sure to send a properly formatted form request body.").WithWrap(err).WithDebug(err.Error()))
+	}
+	request.Form = r.Form
+	request.State = request.Form.Get("state")
```

Comment:
The lines from the start of the function until here are more or less a duplicate of newAuthorizeRequest. Do we need to copy all of the code, or can we re-use newAuthorizeRequest instead?

Reply:
We need the `request` to perform other checks before we call into newAuthorizeRequest. The order of validation follows the spec.

Reply:
ok, make ssnese - thanks!